### PR TITLE
Clean-up Media Type Info

### PR DESCRIPTION
**Pull request by cholmes**
**Related Issue(s):** #930 

**Proposed Changes:**

1. This organizes how we talk about media types a bit better. It makes the media types of stac structures (item/catalog/collection) more clear, moves the common media types to 'best practices', and talks about asset media types a bit differently than link media types.

**PR Checklist:**

- [x] This PR is made against the dev branch (all proposed changes except releases should be against dev, not master).
- [x] This PR has **no** breaking changes.
- [x] a CHANGELOG entry is not required.

### CODE REVIEW

**Comment by cholmes** [comment]
Should probably squash commits on merge, since this has the history of #944 in it.

File: item-spec/item-spec.md [hunk 1]
```diff
@@ -251,6 +228,13 @@ in the file).
 For examples of fields that this construct is recommended for, see the [section of STAC Best Practices](../best-practices.md#common-use-cases-of-additional-fields-for-assets)
 that talks about common use cases of additional fields for assets.
 
+## Media Type for STAC Item
+
+A STAC Item is a GeoJSON file ([RFC 7946](https://tools.ietf.org/html/rfc7946)), and thus should use the 
+`[application/geo+json](https://tools.ietf.org/html/rfc7946#section-12)` as the [Media Type](https://en.wikipedia.org/wiki/Media_type) 
+(previously known as the MIME Type). STAC Items are recommended to add the parameter `profile` with the value `stac-item` to
```
Comment (m-mohr) [inline item-spec/item-spec.md:235]: This is the same issue as with GeoTiff. You can't add parameters, I think it should just be the GeoJSON media type. Also, I think we should say that `application/json` can also occur in implementations. I don't think everybody implemented the GeoJSON media type.

File: item-spec/item-spec.md [hunk 2]
```diff
@@ -167,12 +167,23 @@ or streamed. It is allowed to add additional fields.
...
+| type        | string    | [Media type](#asset-media-type) of the asset. See the [common media types](../best-practices.md#common-media-types-in-stac) in the best practice doc for commonly used asset types. |
 | roles       | \[string] | The [semantic roles](#asset-role-types) of the asset, similar to the use of `rel` in links. |
 
 Beyond to the fields above, any of the [additional fields](#additional-fields) *may* be added to the assets. But this
 is recommended only in special cases, see [Additional Fields for Assets](#additional-fields-for-assets)) for more information.
 
+#### Asset Media Type
+
+Any media type can be used in an Item's asset `type` field, and [registered](https://www.iana.org/assignments/media-types/media-types.xhtml) 
+Media Types are preferred. In cases where custom vendor-specific media types are necessary, they should
```
Comment (m-mohr) [inline item-spec/item-spec.md:179]: I think the .vnd info is a bit out of scope here. Most people will not make up their own media types and register them.

File: catalog-spec/catalog-spec.md [hunk 3]
```diff
@@ -77,6 +77,30 @@ There are additional `rel` types in the [Using Relation Types](../best-practices
...
+
+The following table lists the Media Types to use for STAC structures.
+
+| Media Type                     | Description                                                  |
+| ------------------------------ | ------------------------------------------------------------ |
+| `application/geo+json`	     | A STAC [Item](../item-spec/README.md)                        |
+| `application/json`             | A STAC [Catalog](#stac-catalog-specification)                |
+| `application/json`             | A STAC [Collection](../collection-spec/README.md)            |
+
+### Media Type for STAC Catalogs
```
Comment (m-mohr) [inline catalog-spec/catalog-spec.md:98]: Maybe switch order of the chapters? After the table the media type definition for STAC Catalogs seems superfluous, but if you first define it and then give an overview table, it seems to have a better order.

File: item-spec/item-spec.md (hunk 1 above)
Comment (cholmes) [reply item-spec/item-spec.md:235]: Ah, meant to remove the 'profile' bit. Good idea on saying application/json is ok too.

File: item-spec/item-spec.md (hunk 2 above)
Comment (cholmes) [reply item-spec/item-spec.md:179]: Good catch, will move this out / link to there.

File: catalog-spec/catalog-spec.md (hunk 3 above)
Comment (cholmes) [reply catalog-spec/catalog-spec.md:98]: Sounds good.

**Comment by cholmes** [comment]
I think I addressed all the comments, so this should be good for final review.

**Review by m-mohr — APPROVED** [review]

**Review by matthewhanson — APPROVED** [review]

**Pull request merged**